packages/hotspot/configure/src/root.jsx

```diff
@@ -0,0 +1,112 @@
+import Main from './main';
+import React from 'react';
+import cloneDeep from 'lodash/cloneDeep';
+import PropTypes from 'prop-types';
+
+class Root extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
```

Comment:
do we need this state? can we remove? i've seen this an a few comps and i think they can all be removed

Reply:
so instead of using state should we have something like

```
constructor(props) {
  super(props);
  this.model = props.model;
  // other lines
}
```

did I got it well?

Reply:
remove state, move handlers to custom element, pass as a prop for rendering in root.jsx. with this we have 2 states one in the custom element and 1 in root, i'm saying only have the state in the custom element and trigger renders when necessary.